platform/android/java/lib/src/org/godotengine/godot/vulkan/VkRenderer.kt

```diff
@@ -48,52 +53,56 @@ import android.view.Surface
  */
 internal class VkRenderer {
 
+	private val pluginRegistry: GodotPluginRegistry = GodotPluginRegistry.getPluginRegistry()
+
 	/**
 	 * Called when the surface is created and signals the beginning of rendering.
 	 */
 	fun onVkSurfaceCreated(surface: Surface) {
-		nativeOnVkSurfaceCreated(surface)
+		GodotLib.newcontext(surface, false)
+		for (plugin in pluginRegistry.getAllPlugins()) {
+			plugin.onVkSurfaceCreated(surface)
+		}
 	}
 
 	/**
 	 * Called after the surface is created and whenever its size changes.
 	 */
 	fun onVkSurfaceChanged(surface: Surface, width: Int, height: Int) {
-		nativeOnVkSurfaceChanged(surface, width, height)
+		GodotLib.resize(width, height)
+		GodotLib.newcontext(surface, false)
```

Comment:
Ah yes, good point! Right now, I'm not handling the surface re-creation properly, so the workaround is to reset the app by calling `newcontext`. Instead it should update the surface in the native code, but I haven't looked into that yet.

The consequence is the app gets restarted when paused and resumed again.

It's important to fix and I'll see if I have time for it in the coming days, but I don't think it's a blocker to merge this PR. In the meantime I'll add a comment in the code and in the PR description.

Reply:
Sounds good. 
For the moment, you could then remove the call to `newcontext` from `onVkSurfaceCreated` since it's always followed by `onVkSurfaceChanged` and leave a comment like you mentioned to restore the proper logic flow when surface recreation is properly handled.